components/raftstore/src/store/util.rs

```diff
@@ -907,9 +930,19 @@ pub fn check_conf_change(
             (ConfChangeType::RemoveNode, PeerRole::Voter) if kind != ConfChangeKind::Simple => {
                 return Err(box_err!("{:?}: can not remove voter directly", cp));
             }
-            (ConfChangeType::RemoveNode, _)
-            | (ConfChangeType::AddNode, PeerRole::Voter)
-            | (ConfChangeType::AddLearnerNode, PeerRole::Learner) => {}
+            (ConfChangeType::RemoveNode, _) => {}
+            (ConfChangeType::AddNode, PeerRole::Voter)
+            | (ConfChangeType::AddLearnerNode, PeerRole::Learner) => {
+                if peer.get_id() == leader.get_id()
```

Comment:
Since we use a standalone method for witness, so here we should check whether conf change is used for updating the witness.